packages/webdriverio/webdriverio.d.ts

```diff
@@ -15,50 +14,65 @@ interface AsyncSelectors {
 
 // Element commands wrapper with Promise
 type ElementAsync = {
-    [K in keyof ElementPromise]: WrapWithPromise<ElementPromise[K], ReturnType<ElementPromise[K]>>
+    [K in keyof ElementPromise]:
+    (...args: Parameters<ElementPromise[K]>) => Promise<ReturnType<ElementPromise[K]>>;
 } & AsyncSelectors;
 
 // Element commands that should not be wrapper with promise
 type ElementStatic = Pick<WebdriverIO.Element, 'addCommand'>
 
 // Browser commands that should be wrapper with Promise
-type BrowserPromise = WdioOmit<WebdriverIO.Browser, 'addCommand' | 'options' | '$' | '$$'>;
+type BrowserPromise = Omit<WebdriverIO.Browser, 'addCommand' | 'options' | '$' | '$$'>;
 
 // Browser commands wrapper with Promise
 type BrowserAsync = {
-    [K in keyof BrowserPromise]: WrapWithPromise<BrowserPromise[K], ReturnType<BrowserPromise[K]>>
+    [K in keyof BrowserPromise]:
+    (...args: Parameters<BrowserPromise[K]>) => Promise<ReturnType<BrowserPromise[K]>>;
 } & AsyncSelectors;
 
 // Browser commands that should not be wrapper with promise
 type BrowserStatic = Pick<WebdriverIO.Browser, 'addCommand' | 'options'>;
 declare namespace WebdriverIOAsync {
     function remote(
-        options?: WebDriver.Options & WebdriverIO.Options,
+        options?: WebdriverIO.RemoteOptions,
```

Comment:
Should be here `WebdriverIOAsync` instead of `WebdriverIO`?

Reply:
RemoteOptions is defined in WebdriverIO, so no.

Reply:
https://github.com/webdriverio/webdriverio/pull/4050/files#diff-20adb64d960c251a4c323b411bec1f91R73